add ignore_empty_value parameter in set ingest processor (#57030)
elastic/elasticsearch: docs/reference/ingest/processors/set.asciidoc

| `field` | yes       | -        | The field to insert, upsert, or update. Supports <<accessing-template-fields,template snippets>>.
| `value` | yes       | -        | The value to be set for the field. Supports <<accessing-template-fields,template snippets>>.
| `override` | no        | true     | If processor will update fields with pre-existing non-null-valued field. When set to `false`, such fields will not be touched.

> | `ignore_empty_value` | no        | `false`  | If `true` and `value` is `null` or empty string(only for <<accessing-template-fields,template snippets>>), the processor quietly exits without modifying the document

Reviewer: ```suggestion
| `ignore_empty_value` | no        | `false`  | If `true` and `value` is a <<accessing-template-fields,template snippet>> that evaluates to `null` or the empty string, the processor quietly exits without modifying the document
```
